File: frontend/src/api/regionalHealth.ts
import { apiClient } from './client'

export type RegionalHealthPoint = {
  period: string
  value: number
  label: string
}

export type RegionalHealthSignalSummary = {
  trend_label: string
  review_priority: string
  confidence: string
  change_percent: number | null
  signal_version: string
  limitations: string[]
}

export type RegionalHealthSearchResponse = {
  module: string
  region: string
  category: string
  source_id: string
  source_name: string
  endpoint: string
  query: string
  retrieval_timestamp: string
  record_count: number
  latest_period: string | null
  latest_value: number | null
  previous_period: string | null
  previous_value: number | null
  signal: RegionalHealthSignalSummary
  records: RegionalHealthPoint[]
  disclaimer: string
}

export async function searchRegionalHealth(
  region: string,
  category: string,
): Promise<RegionalHealthSearchResponse> {
  const response = await apiClient.get<RegionalHealthSearchResponse>(
    '/api/v1/regional-health/search',
    {
      params: {
        region,
        category,
      },
    },
  )

  return response.data
}

File: backend/app/schemas/regional_health.py
from pydantic import BaseModel, Field


class RegionalHealthPoint(BaseModel):
    period: str
    value: int
    label: str


class RegionalHealthSignalSummary(BaseModel):
    trend_label: str
    review_priority: str
    confidence: str
    change_percent: float | None = None
    signal_version: str
    limitations: list[str]


class RegionalHealthSearchResponse(BaseModel):
    module: str = "RegionalHealthPulse"
    region: str
    category: str
    source_id: str
    source_name: str
    endpoint: str
    query: str
    retrieval_timestamp: str
    record_count: int
    latest_period: str | None = None
    latest_value: int | None = None
    previous_period: str | None = None
    previous_value: int | None = None
    signal: RegionalHealthSignalSummary
    records: list[RegionalHealthPoint] = Field(default_factory=list)
    disclaimer: str
